polygeist/tools/cgeist/driver.cc

```diff
@@ -317,7 +325,7 @@ static int canonicalize(mlir::MLIRContext &context,
   optPM.addPass(polygeist::createLoopRestructurePass());
   optPM.addPass(polygeist::replaceAffineCFGPass());
   optPM.addPass(mlir::createCanonicalizerPass(canonicalizerConfig, {}, {}));
-  if (ScalarReplacement)
+  if (ScalarReplacement && IsFunc)
```

Comment:
So several optimization will only be performed when the function is a `func::FuncOp` and not when the function is a GPU function ?  Why is that the case and do you plan to remove this limitation later ?

Reply:
This pass is only implemented for `func::FuncOp`. It's a builtin pass. I guess we can port it to also support GPUFuncOp. Will look into it.